subsys/net/lib/nrf_cloud/src/nrf_cloud_credentials.c

```diff
@@ -6,9 +6,77 @@
 #include <net/nrf_cloud.h>
 #include <zephyr/logging/log.h>
 #include <modem/modem_key_mgmt.h>
+#include <zephyr/net/tls_credentials.h>
 
 LOG_MODULE_REGISTER(nrf_cloud_credentials, CONFIG_NRF_CLOUD_LOG_LEVEL);
 
+enum nrf_cloud_cred_type {
+	NRF_CLOUD_CRED_CA_CERT,
+	NRF_CLOUD_CRED_CLIENT_CERT,
+	NRF_CLOUD_CRED_PRIVATE_KEY
+};
+
+/**
+ * @brief Check whether a specified credential exists in any enabled credentials subsystem.
+ *
+ * @retval 0 if the specified credential exists.
+ * @retval -EIO Error while checking if specified credential exists.
+ * @retval -ENOTSUP if the specified  credential does not exist.
+ */
+static int nrf_cloud_credential_check(uint32_t sec_tag, enum nrf_cloud_cred_type cred_type) {
+	enum modem_key_mgmt_cred_type modem_type;
+	enum tls_credential_type tls_type;
+	char *debug_tag;
+	int ret;
+	bool exists;
+
+	/* Translate universal credential type into credential subsystem types. */
+	switch(cred_type) {
+	case NRF_CLOUD_CRED_CA_CERT:
+		modem_type = MODEM_KEY_MGMT_CRED_TYPE_CA_CHAIN;
+		tls_type = TLS_CREDENTIAL_CA_CERTIFICATE;
+		debug_tag = "CA cert";
+		break;
+	case NRF_CLOUD_CRED_CLIENT_CERT:
+		modem_type = MODEM_KEY_MGMT_CRED_TYPE_PUBLIC_CERT;
+		tls_type = TLS_CREDENTIAL_SERVER_CERTIFICATE;
+		debug_tag = "client cert";
+		break;
+	case NRF_CLOUD_CRED_PRIVATE_KEY:
+		modem_type = MODEM_KEY_MGMT_CRED_TYPE_PRIVATE_CERT;
+		tls_type = TLS_CREDENTIAL_PRIVATE_KEY;
+		debug_tag = "private key";
+		break;
+	default:
+		/* Invalid credential type specified. */
+		return -EIO;
+	}
+
+	/* Check each enabled credentials subsystem for the credential. */
+
+#if defined(CONFIG_MODEM_KEY_MGMT)
+	/* Modem credentials subsystem */
+	ret = modem_key_mgmt_exists(sec_tag, modem_type, &exists);
+	if (ret < 0) {
+		LOG_ERR("modem_key_mgmt_exists() failed for %s in sec tag %u, error: %d",
+			debug_tag, sec_tag, ret);
+		return -EIO;
+	}
+	if (exists) {
+		return 0;
```

Comment:
add log msg if `CONFIG_TLS_CREDENTIALS` is also enabled... so user knows that the modem credentials were checked, not native.